docs/reference/anatomy-of-a-recipe.rst

```diff
@@ -0,0 +1,329 @@
+<https://snapcraft.io/docs/snapcraft-yaml-schema#p-21225-app-directives>`_
+define the command path for each app, how it will be run, optional parameters,
+and the interface connections that will be established at runtime.
+
+The yt-dlp recipe declares a single app, which is the main app itself. Other
+snaps may have multiple sub-apps or executables.
+
+The ``command`` key defines the path to the executable -- relative to the snap
+-- and arguments to use when the app runs.
+
+The ``plugs`` key defines the list of interfaces to which the app will have
+access to. This enables the intended app functionality. In this specific case,
+the yt-dlp snap will be allowed access to the home, network and removable-media
+interfaces, which are not available by default under strict confinement. This
+will allow the user of the tool to access files in the user's home directory,
+from a network connection, or from any mounted removable media locations.
+
+
+Advanced recipe
+---------------
+
+For a more complex example, there's the recipe for wethr, a CLI command for
+retrieving local weather conditions.
+
+The metadata, base, and confinement declarations are rather similar to the
+simple example, but with some notable differences.
+
+.. collapse:: wethr recipe
+
+  .. literalinclude:: code/wethr-recipe.yaml
+    :language: yaml
+    :lines: 2-
+
+.. note::
+
+  The recipe shown here has been modified from the actual snap's recipe to
+  highlight features of Snapcraft.
+
+
+Adopting metadata
+~~~~~~~~~~~~~~~~~
+
+The ``adopt-info`` key instructs Snapcraft to import metadata from another
+source. Such use can be useful for continuous integration and delivery systems,
+where the declarations in the recipe can be obtained from scripts rather than
+manually.
+
+There are multiple ways that information can be obtained. For a how-to guide on
+bringing in external metadata, see `Using external metadata
+<https://snapcraft.io/docs/using-external-metadata>`_.
+
+Multiple metadata fields can be populated using this key. In this recipe, the
+snap's version is obtained from the Git repository release tag, which proceeds
+in two stages:
+
+#. The ``adopt-info`` key instructs Snapcraft to populate the metadata fields
+   that aren't already declared in the recipe.
+#. In the parts section at the end of the recipe:
+   #. A step in the build lifecycle is manually overridden.
+   #. A custom script is used to derive the version string.
+   #. The version string is set using the Snapcraftctl scriptlet.
+
+Alternatively, in this particular example, the version field could also be
+manually set with ``version: '1.5'``.
+
+
+Quality grade
+~~~~~~~~~~~~~
+
+The ``grade`` key defines the quality level of the snap. Two levels are
+available, devel and stable. Snaps with the devel grade can't be uploaded to
+either of the stable or candidate channels in the Snap Store.
+
+
+Architectures
+~~~~~~~~~~~~~
+
+The ``architectures`` key defines the target platforms for which the snap
```

Comment:
You could link to :doc:`/reference/architectures` :sunglasses:

Reply:
1df1efa25